include/datadog/telemetry/telemetry.h

```diff
@@ -16,19 +16,40 @@ class TracerTelemetry;
 
 namespace telemetry {
 
-class Telemetry {
+/// The telemetry class is responsible for handling internal telemetry data to
+/// track Datadog product usage. It _can_ collect and report logs and metrics.
+///
```

Comment:
Don't we have private headers ? If yes, is there a reason to put the file in the public headers and say do not use ?